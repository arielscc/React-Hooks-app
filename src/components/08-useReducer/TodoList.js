import React from 'react';
import PropTypes from 'prop-types';
import TodoItem from './TodoItem';

const TodoList = React.memo(({ todos, handleToggle, handleDelete }) => {
  return (
    <ul>
      {todos.map((todo, i) => (
        <TodoItem
          key={todo.id}
          todo={todo}
          index={i}
          handleToggle={handleToggle}
          handleDelete={handleDelete}
        />
      ))}
    </ul>
  );
});

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

export default TodoList;
