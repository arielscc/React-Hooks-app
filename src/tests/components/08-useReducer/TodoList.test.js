import { shallow } from 'enzyme';
import React from 'react';
import TodoList from '../../../components/08-useReducer/TodoList';
import { demoTodos } from '../../fixtures/demoTodos';

const handleToggle = jest.fn();
const handleDelete = jest.fn();

describe('Pruebas en el componente TodoList', () => {
  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />
  );

  test('Debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de tener 2 <TodoItem/>', () => {
    expect(wrapper.find('TodoItem').length).toBe(demoTodos.length);
    expect(wrapper.find('TodoItem').at(0).prop('handleDelete')).toEqual(
      expect.any(Function)
    );
  });
});
