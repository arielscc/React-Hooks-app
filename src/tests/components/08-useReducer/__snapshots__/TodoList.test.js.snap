// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente TodoList Debe de mostrarse correctamente 1`] = `
<ul>
  <TodoItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "done": false,
        "id": 1,
        "todo": "Aprender react",
      }
    }
  />
  <TodoItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "done": false,
        "id": 2,
        "todo": "Aprender Mongo",
      }
    }
  />
</ul>
`;
